File: app/src/main/java/com/example/msdc/adapter/ImageAdapter.java
package com.example.msdc.adapter;

import android.widget.ImageView;
import android.widget.Toast;

import com.squareup.picasso.Callback;
import com.squareup.picasso.Picasso;

public class ImageAdapter {
	public static final String POSTER_IMAGE_BASE_URL = "https://image.tmdb.org/t/p/w500";

	public static final String POSTER_IMAGE_LOGO_URL = "https://image.tmdb.org/t/p/w185";
	
	public static final String PROFILE_IMAGE_BASE_URL = "https://image.tmdb.org/t/p/w185";

    public static final String BACKDROP_IMAGE_BASE_URL = "https://image.tmdb.org/t/p/original";
	
	public static final String SEARCH_IMAGE_BASE_URL = "https://image.tmdb.org/t/p/w154";


    public static void setPosterURL(ImageView imageView, String path){
        try{
            imageView.setAlpha(0f);
            Picasso.get().load(POSTER_IMAGE_BASE_URL + path ).noFade().into(imageView, new Callback(){
                @Override
                public void onSuccess(){imageView.animate().setDuration(500).alpha(1f).start();}

                @Override
                public void onError(Exception e){
                    Toast.makeText(imageView.getContext(), e.getMessage() + " cause : "
                            + e.getCause(), Toast.LENGTH_SHORT).show();
                }
            });
        } catch(Exception ignored){
        }
    }

    public static void setPosterLogoURL(ImageView imageView, String path){
        try{
            imageView.setAlpha(0f);
            Picasso.get().load(POSTER_IMAGE_LOGO_URL + path ).noFade().into(imageView, new Callback(){
                @Override
                public void onSuccess(){imageView.animate().setDuration(500).alpha(1f).start();}

                @Override
                public void onError(Exception e){
                    Toast.makeText(imageView.getContext(), e.getMessage() + " cause : "
                            + e.getCause(), Toast.LENGTH_SHORT).show();
                }
            });
        } catch(Exception ignored){
        }
    }

    public static void setProfileLogoURL(ImageView imageView, String path){
        try{
            imageView.setAlpha(0f);
            Picasso.get().load(PROFILE_IMAGE_BASE_URL + path ).noFade().into(imageView, new Callback(){
                @Override
                public void onSuccess(){imageView.animate().setDuration(500).alpha(1f).start();}

                @Override
                public void onError(Exception e){
                    Toast.makeText(imageView.getContext(), e.getMessage() + " cause : "
                            + e.getCause(), Toast.LENGTH_SHORT).show();
                }
            });
        } catch(Exception ignored){
        }
    }

    public static void setSearchImageBaseURL(ImageView imageView, String path){
        try{
            imageView.setAlpha(0f);
            Picasso.get().load(SEARCH_IMAGE_BASE_URL + path ).noFade().into(imageView, new Callback(){
                @Override
                public void onSuccess(){imageView.animate().setDuration(500).alpha(1f).start();}

                @Override
                public void onError(Exception e){
                    Toast.makeText(imageView.getContext(), e.getMessage() + " cause : "
                            + e.getCause(), Toast.LENGTH_SHORT).show();
                }
            });
        } catch(Exception ignored){
        }
    }

    public static void setBackdropURL(ImageView imageView, String path){
        try{
            imageView.setAlpha(0f);
            Picasso.get().load(BACKDROP_IMAGE_BASE_URL + path ).noFade().into(imageView, new Callback(){
                @Override
                public void onSuccess(){imageView.animate().setDuration(500).alpha(1f).start();}

                @Override
                public void onError(Exception e){
                    Toast.makeText(imageView.getContext(), e.getMessage() + " cause : "
                            + e.getCause(), Toast.LENGTH_SHORT).show();
                }
            });
        } catch(Exception ignored){
        }
    }
}

File: app/src/main/java/com/example/msdc/adapter/ImageAdapter.kt
package com.example.msdc.adapter

import android.widget.ImageView
import android.widget.Toast
import com.squareup.picasso.Callback
import com.squareup.picasso.Picasso

object ImageAdapter {
    private const val POSTER_IMAGE_LOGO_URL = "https://image.tmdb.org/t/p/w185"
    private const val PROFILE_IMAGE_BASE_URL = "https://image.tmdb.org/t/p/w185"
    private const val POSTER_IMAGE_BASE_URL = "https://image.tmdb.org/t/p/w500"
    private const val POSTER_HD_BASE_URL = "https://image.tmdb.org/t/p/w780"
    private const val BACKDROP_IMAGE_BASE_URL = "https://image.tmdb.org/t/p/original"

    @JvmStatic
    fun setPosterURL(imageView: ImageView, path: String?) {
        try {
            imageView.alpha = 0f
            Picasso.get().load(POSTER_IMAGE_BASE_URL + path).noFade()
                .into(imageView, object : Callback {
                    override fun onSuccess() {
                        imageView.animate().setDuration(500).alpha(1f).start()
                    }

                    override fun onError(e: Exception) {
                        Toast.makeText(
                            imageView.context, e.message + " cause : "
                                    + e.cause, Toast.LENGTH_SHORT
                        ).show()
                    }
                })
        } catch (ignored: Exception) {
        }
    }

    @JvmStatic
    fun setPosterLogoURL(imageView: ImageView, path: String?) {
        try {
            imageView.alpha = 0f
            Picasso.get().load(POSTER_IMAGE_LOGO_URL + path).noFade()
                .into(imageView, object : Callback {
                    override fun onSuccess() {
                        imageView.animate().setDuration(500).alpha(1f).start()
                    }

                    override fun onError(e: Exception) {
                        Toast.makeText(
                            imageView.context, e.message + " cause : "
                                    + e.cause, Toast.LENGTH_SHORT
                        ).show()
                    }
                })
        } catch (ignored: Exception) {
        }
    }

    fun setProfileLogoURL(imageView: ImageView, path: String?) {
        try {
            imageView.alpha = 0f
            Picasso.get().load(PROFILE_IMAGE_BASE_URL + path).noFade()
                .into(imageView, object : Callback {
                    override fun onSuccess() {
                        imageView.animate().setDuration(500).alpha(1f).start()
                    }

                    override fun onError(e: Exception) {
                        Toast.makeText(
                            imageView.context, e.message + " cause : "
                                    + e.cause, Toast.LENGTH_SHORT
                        ).show()
                    }
                })
        } catch (ignored: Exception) {
        }
    }

    fun setPosterHD(imageView: ImageView, path: String?) {
        try {
            imageView.alpha = 0f
            Picasso.get().load(POSTER_HD_BASE_URL + path).noFade()
                .into(imageView, object : Callback {
                    override fun onSuccess() {
                        imageView.animate().setDuration(500).alpha(1f).start()
                    }

                    override fun onError(e: Exception) {
                        Toast.makeText(
                            imageView.context, e.message + " cause : "
                                    + e.cause, Toast.LENGTH_SHORT
                        ).show()
                    }
                })
        } catch (ignored: Exception) {
        }
    }

    @JvmStatic
    fun setBackdropURL(imageView: ImageView, path: String?) {
        try {
            imageView.alpha = 0f
            Picasso.get().load(BACKDROP_IMAGE_BASE_URL + path).noFade()
                .into(imageView, object : Callback {
                    override fun onSuccess() {
                        imageView.animate().setDuration(500).alpha(1f).start()
                    }

                    override fun onError(e: Exception) {
                        Toast.makeText(
                            imageView.context, e.message + " cause : "
                                    + e.cause, Toast.LENGTH_SHORT
                        ).show()
                    }
                })
        } catch (ignored: Exception) {
        }
    }
}